src/vsr/client.zig

```diff
@@ -205,65 +217,70 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
             message_body_size: usize,
         ) void {
             assert(@enumToInt(operation) >= constants.vsr_operations_reserved);
-
-            self.register();
-            assert(self.request_number > 0);
+            assert(message_body_size <= constants.message_body_size_max);
 
             // We will set parent, context, view and checksums only when sending for the first time:
             message.header.* = .{
                 .client = self.id,
-                .request = self.request_number,
                 .cluster = self.cluster,
                 .command = .request,
                 .operation = vsr.Operation.from(StateMachine, operation),
                 .size = @intCast(u32, @sizeOf(Header) + message_body_size),
             };
+            self.raw_request(user_data, .{ .request = callback }, message);
+        }
 
-            log.debug("{}: request: user_data={} request={} size={} {s}", .{
-                self.id,
-                user_data,
-                message.header.request,
-                message.header.size,
-                @tagName(operation),
-            });
-
-            assert(!self.request_queue.full());
-            const was_empty = self.request_queue.empty();
-
-            self.request_number += 1;
-            self.request_queue.push_assume_capacity(.{
-                .user_data = user_data,
-                .callback = callback,
-                .message = message.ref(),
-            });
-            if (self.request_queue.full()) assert(self.messages_available == 0);
+        pub fn request_reconfigure(
```

Comment:
> For the client, we go with a strongly typed API -- .requset and .request_reconfigure are different functions, and they accept callbacks with different signatures. I am not entirely sure that's a good idea --- as an alternative, we could use more "dynamically-typed" API, where it is just a request, and its on the user to cast the result to ReconfigurationResult.

My first response is that it feels a bit inconsistent to use a separate API for just reconfiguration. 

:thinking: Does the `Client` needs to "know" about reconfigurations, the same way that it needs to "know" about `register`s? i.e. when a client sends/receives a reconfiguration does the client itself need to take some special action that it wouldn't take for a different message?